File: Demo/Blazor/Blazor.Server/ViewModels/Form/AddressForm.cs
using System.Linq;
using System.Reactive.Linq;
using DotNetify;
using DotNetify.Elements;

namespace Blazor.Server
{
   public class AddressForm : BaseVM
   {
      public ReactiveProperty<Customer> Customer { get; } = new ReactiveProperty<Customer>();

      public AddressForm()
      {
         AddProperty<string>(nameof(AddressInfo.Address1))
            .WithAttribute(new TextFieldAttribute { Label = "Address 1:" })
            .SubscribeTo(Customer.Select(x => x.Address.Address1));

         AddProperty<string>(nameof(AddressInfo.Address2))
            .WithAttribute(new TextFieldAttribute { Label = "Address 2:" })
            .SubscribeTo(Customer.Select(x => x.Address.Address2));

         AddProperty<string>(nameof(AddressInfo.City))
            .WithAttribute(new TextFieldAttribute { Label = "City:" })
            .SubscribeTo(Customer.Select(x => x.Address.City));

         AddProperty<State>(nameof(AddressInfo.State))
            .WithAttribute(new DropdownListAttribute
            {
               Label = "State:",
               Options = typeof(State).ToDescriptions()
            })
            .SubscribeTo(Customer.Select(x => x.Address.State));

         AddProperty<string>(nameof(AddressInfo.ZipCode))
            .WithAttribute(new TextFieldAttribute { Label = "Zip Code:" })
            .SubscribeTo(Customer.Select(x => x.Address.ZipCode));
      }
   }
}

File: _experimental/Mfe/App1/client/form/AddressForm.js
import React from 'react';
import { Cell, DropdownList, Form, NumberField, Panel, TextField, VMContext } from 'dotnetify-elements';

const AddressForm = () => (
  <Panel>
    <Cell header="Primary Address">
      <VMContext vm="AddressForm">
        <Form id="Address">
          <Panel flex childProps={{ horizontal: true }}>
            <TextField id="Address1" />
            <TextField id="Address2" />
            <TextField id="City" />
            <DropdownList id="State" />
            <NumberField id="ZipCode" />
          </Panel>
        </Form>
      </VMContext>s
    </Cell>
  </Panel>
);

export default AddressForm;
